### DISCUSSION TITLE
Proxmox VE Monitor-All starts messes up with boot order options

### DISCUSSION THREAD

**Discussion by sammy708** [tteck/Proxmox#2370] [Help Needed]
### Please verify that you have read and understood the guidelines.

Yes

### A clear and concise description of the issue.

Good morning, just noticed after a PVE reboot, that the monitor-all service starts a not running VM right at the beginning, although according to boot order option one may want postpone the start of a VM for some time. E.g. boot homeassistant some minutes after startup of Maria DB.
Any idea how to configure a initial waiting time, until the monitor-all service starts monitoring the VMs/Containers?
Thanks and take care! Sam

### Which Linux distribution are you employing?

Default Settings

### Please provide detailed steps to reproduce the issue.

Reboot PVE and observe var/log/ping-instances.log for VMs getting started although according to boot delay they shouldn’t yet.

**Comment by tteck** [comment]
> - For suggestions, questions or feature/script requests, please share them in the [Discussions section.](https://github.com/tteck/Proxmox/discussions) 

Moving to Discussions.

**Comment by tteck** [comment]
Maybe add a `ExecStartPre=/bin/sleep 300` before the `ExecStart=/usr/local/bin/ping-instances.sh` in the service file?

**Comment by sammy708** [comment]
Thanks, I feel like systemd doesn't really want me to do this. With a longer delay this causes timeouts and even with shorter delay `systemctl start ping-instances` doesn't return until the sleep has completed.

I solved it by modifing `/etc/systemd/system/ping-instances.timer` and increased `OnBootSec` to 600:

`[Timer]`
`OnBootSec=600`

and ensured the service is not started on boot automaticaly, but only triggered by the timer with the followint command:
`systemctl disable ping-instances`

**Reply by tteck** [reply]
I had forgotten about adding a timer service. 🤭
